pkg/client/block/client.go

```diff
@@ -58,20 +73,80 @@ type blockClient struct {
 	// These enable the EventsReplayClient to correctly map the raw event bytes
 	// to Block objects and to correctly return a BlockReplayObservable
 	eventsReplayClient client.EventsReplayClient[client.Block]
+	// latestBlockMu is a mutex that protects the latestBlock field from concurrent
+	// writes from getLatestBlocks and getInitialBlock.
+	latestBlockMu *sync.Mutex
+	// latestBlock is the last block observed by the blockClient.
+	// It is initialized by the getInitialBlock function and then updated by the
+	// updateLatestBlock goroutine that listens to the eventsReplayClient.
+	latestBlock client.Block
 }
 
 // CommittedBlocksSequence returns a replay observable of new block events.
 func (b *blockClient) CommittedBlocksSequence(ctx context.Context) client.BlockReplayObservable {
 	return b.eventsReplayClient.EventsSequence(ctx)
 }
 
-// LatestsNBlocks returns the last n blocks observed by the BockClient.
+// LastNBlocks returns the last n blocks observed by the BlockClient.
 func (b *blockClient) LastNBlocks(ctx context.Context, n int) []client.Block {
 	return b.eventsReplayClient.LastNEvents(ctx, n)
 }
 
+// LastBlock returns the last blocks observed by the BlockClient.
+func (b *blockClient) LastBlock() client.Block {
+	b.latestBlockMu.Lock()
+	defer b.latestBlockMu.Unlock()
+	return b.latestBlock
+}
+
 // Close closes the underlying websocket connection for the EventsQueryClient
 // and closes all downstream connections.
 func (b *blockClient) Close() {
 	b.eventsReplayClient.Close()
 }
+
+// getLatestBlocks listens to the EventsReplayClient's EventsSequence and updates
```

Comment:
Shold we rename to `goListenForLatestBlock`

1. It's just one block, not `s`
2. It's intended to run as a goroutine. Make sure to update the comments
3. It doesn't just get, it constantly listens for it.